lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -191,37 +191,38 @@ private static function get_settings() {
 
 		foreach ( $variations as $variation ) {
 
-			// Skip if settings.typography.fontFamilies are not defined in the variation.
-			if ( empty( $variation['settings']['typography']['fontFamilies'] ) ) {
+			// Skip if settings.typography.fontFamilies.theme is not defined or is not an array.
+			if (
+				! isset( $variation['settings']['typography']['fontFamilies']['theme'] ) ||
+				! is_array( $variation['settings']['typography']['fontFamilies']['theme'] )
+			) {
 				continue;
 			}
 
 			// One time, set any missing parts of the array structure.
 			if ( $set_theme_structure ) {
 				$set_theme_structure = false;
 				$settings            = static::set_tyopgraphy_settings_array_structure( $settings );
-			}
 
-			// Initialize the font families from variation if set and is an array, otherwise default to an empty array.
-			$variation_font_families = ( isset( $variation['settings']['typography']['fontFamilies']['theme'] ) && is_array( $variation['settings']['typography']['fontFamilies']['theme'] ) )
-				? $variation['settings']['typography']['fontFamilies']['theme']
-				: array();
+				// Font families from settings must be an array.
+				if (
+					! isset( $settings['typography']['fontFamilies']['theme'] ) ||
+					! is_array( $settings['typography']['fontFamilies']['theme'] )
+				) {
+					$settings['typography']['fontFamilies']['theme'] = array();
```

Comment:
I think we don't need this anymore as the `set_tyopgraphy_settings_array_structure` sets the `typography.fontFamilies.theme` structure in the given array and it works well without this check now

Reply:
Hmm, thinking that `set_tyopgraphy_settings_array_structure()` should be ensuring this is an array, not just checking if empty. I.e. move this code:
```
if (
	! isset( $settings['typography']['fontFamilies']['theme'] ) ||
	! is_array( $settings['typography']['fontFamilies']['theme'] )
) {
	$settings['typography']['fontFamilies']['theme'] = array();
}
```
 from `get_settings()` to `set_tyopgraphy_settings_array_structure()`.

Reply:
Sounds good!